react/week3/todo-week3/src/components/ToDoList.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { ToDoItemRow } from "./ToDoItemRow";
+import "../App.css";
+//import PropTypes from "prop-types";
+
+class ToDoList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toDoList: [],
+      newTodo: "",
+      newTodoDeadline: "",
+    };
+    this.fetchData = this.fetchData.bind(this);
+    this.addItem = this.addItem.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.changeHandler = this.changeHandler.bind(this);
+  }
+  componentDidMount() {
+    this.fetchData();
+  }
+  fetchData() {
```

Comment:
When you have functions like this that do two things make sure you either describe that it is doing two things in the title or seperate the things it is doing. 

So in componentDidMount you could have 

```
const todoItems = await this.fetchTodos();
this.setState({
    toDoList: todoItems ,
})
```

I would definitely recommend the second solution. Seperating it out

```suggestion
  fetchAndUpdateTodos() {
```